dima/react-movies/src/pages/mainpage/mainpage.tsx

```diff
@@ -0,0 +1,199 @@
+          ...defaultParams,
+          ...oldParamsObj,
+          sortBy: oldParamsObj.sortBy ? oldParamsObj.sortBy : currentSortType,
+          offset: pageNum * 9,
+        });
+        break;
+    }
+  };
+
+  switchCurrentSortType = (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
+  ): void => {
+    const { location, history, currentSortType } = this.props;
+    const oldParam = QueryString.parse(location.search);
+    const { value } = e.currentTarget;
+    if (
+      !oldParam.sortBy ||
+      (oldParam.sortBy !== value && currentSortType !== value)
+    ) {
+      const newParamString = QueryString.stringify({
+        ...oldParam,
+        sortBy: value,
+      });
+      history.push({
+        pathname: location.pathname,
+        search: newParamString,
+      });
+      this.props.setCurrentSortType(
+        value === SortProperty.date ? SortProperty.date : SortProperty.rating
+      );
+    }
+  };
+  getPage = (): number => {
+    const { history } = this.props;
+    const params = QueryString.parse(history.location.search);
+    const numPage =
+      params.page && Number(params.page.toString()) - 1 >= 0
+        ? Number(params.page.toString()) - 1
+        : 0;
+    return numPage;
+  };
+  onPageChanged = (selected: number): void => {
+    const { history } = this.props;
+    const params = QueryString.parse(history.location.search);
+    history.push({
+      pathname: history.location.pathname,
+      search: QueryString.stringify({ ...params, page: selected + 1 }),
+    });
+  };
+  renderPanel = (): React.ReactNode => {
+    const { movies, currentSortType, loading } = this.props;
+    return (
+      <>
+        <ErrorBoundary>
+          <MoviesResult total={movies.total} loading={loading} />
+        </ErrorBoundary>
+        <ErrorBoundary>
+          <SortFilter
+            currentSortType={currentSortType}
+            switchCurrentSortType={this.switchCurrentSortType}
+          />
+        </ErrorBoundary>
+      </>
+    );
+  };
+
+  render() {
+    const { movies, loading } = this.props;
+    const { data, total } = movies;
+
+    return (
+      <div className="app">
+        <div className="first-screen-wrapper">
+          <ContentContainer>
+            <Header />
+            <ErrorBoundary>
+              <SearchBar
+                location={this.props.location}
+                pushParams={this.pushParams}
+              />
+            </ErrorBoundary>
+          </ContentContainer>
+        </div>
+        <div className="second-screen-wrapper">
+          <ContentContainer>
+            <div className="flex-wrapper">
+              {data?.length !== 0 && this.renderPanel()}
+            </div>
+          </ContentContainer>
+        </div>
+        <div className="third-screen-wrapper">
+          <ContentContainer>
+            <ErrorBoundary>
+              <MoviesContainer
+                movies={data}
+                total={total}
+                getPage={this.getPage}
```

Comment:
Why do you pass getPage func?? Just pass a result of it and call it page 
`page={this.getPage()}`